src/gennylib/src/quiesce.cpp

```diff
@@ -94,6 +94,65 @@ bool waitOplog(v1::Topology& topology) {
     return waitVisitor.success();
 }
 
+bool CheckForDroppedCollections(v1::Topology& topology, std::string dbName, SleepContext sleepContext) {
+    class CheckForDroppedCollectionsVisitor : public v1::TopologyVisitor {
+    public:
+        CheckForDroppedCollectionsVisitor(std::string dbName, SleepContext sleepContext)
+            : _dbName{dbName},
+              _sleepContext{sleepContext} {}
+
+        void onBeforeReplSet(const v1::ReplSetDescription& desc) override {
+            mongocxx::client client(mongocxx::uri(desc.primaryUri));
+            _successAcc = checkCollectionsTestDB(client);
+        }
+
+        bool success() {
+            return _successAcc;
+        }
+
+    private:
+        bool checkCollections(mongocxx::database db) {
+            using bsoncxx::builder::basic::kvp;
+            using bsoncxx::builder::basic::make_document;
+
+            auto res = db.run_command(make_document(kvp("serverStatus", 1)));
+            if (!res.view()["storageEngine"] || !res.view()["storageEngine"]["dropPendingIdents"]) {
```

Comment:
We call this method on all of the mongo servers (mongos, mongod, configsvr), but only the mongod's themselves have collections that needed to be dropped